plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergSystemTables.java

```diff
@@ -236,5 +245,67 @@ public void testFilesTable()
                         "('split_offsets', 'array(bigint)', '', '')," +
                         "('equality_ids', 'array(integer)', '', '')");
         assertQuerySucceeds("SELECT * FROM test_schema.\"test_table$files\"");
+
+        MaterializedResult result = computeActual("SELECT file_format, record_count, lower_bounds, upper_bounds FROM test_schema.\"test_table$files\" order by file_path");
+
+        MaterializedResult expectedStatistics =
+                MaterializedResult.resultBuilder(getSession(), VARCHAR, BIGINT, new MapType(INTEGER, VARCHAR, new TypeOperators()),
+                                new MapType(IntegerType.INTEGER, VARCHAR, new TypeOperators()))
+                        .row("ORC", 1L, Map.of(Integer.valueOf(1), "0", Integer.valueOf(2), "2019-09-08"), Map.of(Integer.valueOf(1), "0", Integer.valueOf(2), "2019-09-08"))
+                        .row("ORC", 2L, Map.of(Integer.valueOf(1), "1", Integer.valueOf(2), "2019-09-09"), Map.of(Integer.valueOf(1), "2", Integer.valueOf(2), "2019-09-09"))
+                        .row("ORC", 1L, Map.of(Integer.valueOf(1), "3", Integer.valueOf(2), "2019-09-09"), Map.of(Integer.valueOf(1), "3", Integer.valueOf(2), "2019-09-09"))
+                        .row("ORC", 2L, Map.of(Integer.valueOf(1), "4", Integer.valueOf(2), "2019-09-10"), Map.of(Integer.valueOf(1), "5", Integer.valueOf(2), "2019-09-10"))
+                        .build();
+
+        assertEquals(result, expectedStatistics);
+    }
+
+    @Test
+    public void testAllFilesTable()
+    {
+        assertQuery("SHOW COLUMNS FROM test_schema.\"test_table$all_files\"",
```

Comment:
Can you use a custom table for this test to showcase that "all_files" metadata table contains ALL valid data files and not only the ones corresponding to the current snapshot?

`insert into my_table`
`check that $files v1 and $all_files are the same`
`insert into my_table`
`check that $files v2 is included in $all_files  and the union of $files v1 and $files v2 is equal to $all_files`

Reply:
@findinpath refactored the test-method.